### ISSUE TITLE
Unexpected CONSTANTEND (structure) - Due to % in name

### ISSUE THREAD

**Issue by ed-holland** [abaplint/abaplint#1281]
Issues:
```
/tmp/abaplint-EFnfbA/src/szme/context_i_unit_conv_factor.prog.abap[59, 1] - Unexpected CONSTANTEND (structure)
/tmp/abaplint-EFnfbA/src/szme/context_i_unit_conversion.prog.abap[118, 1] - Unexpected CONSTANTEND (structure)
```
Abap:
![image](https://user-images.githubusercontent.com/53995991/92358831-6a9e6380-f0ea-11ea-9a45-3aeeee46ad62.png)
![image](https://user-images.githubusercontent.com/53995991/92358864-7be77000-f0ea-11ea-8a4f-a4a0a4496add.png)

**Comment by larshp** [comment]
hmm, what does `%_PREDEFINED` do?

![image](https://user-images.githubusercontent.com/5888506/92382945-c4645500-f10d-11ea-8780-8941cbaf97a5.png)

**Comment by larshp** [comment]
also looks like `DATA foo TYPE c %_PREDEFINED.` is valid syntax

**Comment by ed-holland** [comment]
Looks like it is internal syntax. Double-Click does not jump.

**Comment by ed-holland** [comment]
Issue is resolved with 2.42.0. Thanks.
